enhancements/network/multi-network-vrfs.md

```diff
@@ -0,0 +1,499 @@
+
+
+![Egress IP VRF SGW](images/egress-ip-vrf-sgw.svg)
+
+
+For layer 2, egress IP has never been supported before. With the IC design, there is no need to have an ovn_cluster_router and join switch separating the layer 2 switch network (transit switch) from the GR. For non-IC cases this might be necessary, but for OpenShift purposes will only describe the behavior of IC in this proposal. In the layer 2 IC model, GRs per node on a network will all be connected to the layer 2 transit switch:
+
+
+![Egress IP Layer 2](images/egress-ip-l2-primary.svg)
+
+
+In the above diagram, Node 2 is chosen to be the egress IP node for any pods in namespace A. Pod 1 and Pod 2 have default gateway routes to their respective GR on their node. When egress traffic leaves Pod 2, it is sent towards its GR-A on node 2, where it is SNAT’ed to the egress IP and the traffic sent to br-ex. For Pod 1, its traffic is sent to its GR-A on Node 1, where it is then rerouted towards GR-A on Node 2 for egress.
+
+
+#### Egress Firewall
+
+Egress firewall is enforced at the OVN logical switch, and this proposal has no effect on its functionality.
+
+
+#### Egress QoS
+
+Egress QoS is namespace scoped and functions by marking packets at the OVN logical switch, and this proposal has no effect on its functionality.
+
+
+#### Egress Service
+
+Egress service is namespace scoped and its primary function is to SNAT egress packets to a load balancer IP. As previously mentioned, the feature works the same in shared and local gateway mode, by leveraging the local gateway mode path. Therefore its design will be covered in the Local Gateway Mode section of the Design Details.
+
+
+#### Multiple External Gateways (MEG)
+
+There will be no support for MEG or pod direct ingress on any network other than the primary default network. This support may be enhanced later by extending VRFs/networks outside of the cluster.
```

Comment:
when you say "pod direct ingress on any network other than the primary default network" what exactly do you mean here ? The cluster default network, or the per namespace "unique" network you're proposing in the enhancement ? 

We (CNV) are keen on having this feature - direct pod ingress - on any arbitrary secondary network.

Reply:
FWIW, after reviewing within my org, we are fine w/ having direct pod access in the primary "user-defined" network.

Reply:
sorry I'll fix this text to say "cluster default network". The issue with providing support for user-defined networks (primary or secondary) is that pod subnets for those networks may overlap. So if packets come into br-ex destined for a pod IP that exists in multiple networks, how do we know which one to send it to? The only way to allow this is with BGP VPN support, where we can advertise vpnv4/v6 routes and extend the VRF out into the provider network. That will come later.